services/harmony/app/frontends/jobs.ts

```diff
@@ -171,6 +172,58 @@ export async function getJobsListing(
   }
 }
 
+/**
+ * Get a message explaining the change in size from the input to the output
+ *
+ * @param sizes - original and output sizes of the input in MiB (1024 x 1024 bytes)
+ * @param precision - the number of decimal places to allow in the output
+ * @returns a message explaining the size change as a percentage
+ */
+export function sizeChangeMessage(
+  sizes: { originalSize: number; outputSize: number; },
+  precision: number = 2): string {
+  if (sizes.originalSize === 0) {
+    return 'Original size is 0 - percent size change N/A';
+  }
+  if (sizes.outputSize === 0) {
+    return 'Output size is 0 - percent size change N/A';
+  }
+  let result: string;
+  const diff = sizes.originalSize - sizes.outputSize;
+  if (diff < 0) {
+    const percent = (-diff / sizes.originalSize * 100.0).toFixed(precision);
+    result = `${percent}% increase`;
+  } else if (diff > 0) {
+    let percent = (diff / sizes.originalSize * 100.0).toFixed(precision);
+    // due to JS precision issues, big changes will appear to be 100% reduction, which is impossible
+    if (percent === '100.00') percent = '99.99';
```

Comment:
nit: This will not work if the precision is not the default (2).

Reply:
Good catch. Fixed and test added.